spec/compiler/formatter/formatter_spec.cr

```diff
@@ -733,7 +719,155 @@ describe Crystal::Formatter do
       end
       CRYSTAL
 
-    assert_format "macro f\n  yield\n  {{ yield }}\nend"
+    assert_format "macro f\n  yield\n  {{ yield }}\nend", flags: %w[method_signature_yield]
+  end
+
+  context "does not add `&` without flag `method_signature_yield`" do
```

Comment:
I'd suggest to automate it within the `assert_format` method.
Checking whether the code doesn't change should be sufficient.

Reply:
> Checking whether the code doesn't change should be sufficient.

Not it's not.
The majority of test cases actually expects the format to change, it's just two different results: with and without the flag.

Reply:
I think that would be too confusing. Having separate examples is fine.
This way it's easy to remove the entire non-flag group when the flag gets enabled.